src/pixiewps.c

```diff
@@ -65,6 +67,89 @@ static const struct option long_options[] = {
 	{  0,          0,                 0,  0  }
 };
 
+#define SECS_PER_JOB_BLOCK 1000
+
+struct crack_job {
+	pthread_t thr;
+	time_t start;
+};
+
+static struct job_control {
+	int jobs;
+	time_t end;
+	uint32_t randr_enonce[4];
+	struct crack_job *crack_jobs;
+	volatile uint32_t print_seed;
+} job_control;
+
+static void* crack_thread(void *arg) {
+	struct crack_job * j = arg;
+	struct random_data *buf = calloc(1, sizeof(struct random_data));
+	char *rand_statebuf = calloc(1, 128);
+
+	uint32_t seed = j->start;
+	uint32_t limit = job_control.end;
+	initstate_r(seed, rand_statebuf, 128, buf);
+	int32_t res = 0;
+
+	while (!job_control.print_seed) {
+		srandom_r(seed, buf);
+		unsigned i;
+		for (i = 0; i < 4; i++) {
+			random_r(buf, &res);
+			if ((uint32_t) res != job_control.randr_enonce[i])
+				break;
+		}
+
+		if (i == 4) {
+			job_control.print_seed = seed;
+			DEBUG_PRINT("Seed found");
+		}
+
+		seed--;
```

Comment:
hmm, since seed is unsigned, if the entire range till the unix epoch is searched (value 0), then this could lead to seed wrapping around. to catch that, i suppose we should add a check if(seed == 0) break; just before this line, to play safe (in the wrap-around situation, force mode may run forever).